src/python_testing/TC_IDM_4_3.py

```diff
@@ -0,0 +1,234 @@
+                # TestStep(21, "TH sends a subscription request to subscribe to all attributes from a specific cluster on all endpoints. AttributePath = [[Cluster = ClusterID]]. Set the MinIntervalFloor to some value say \"N\"(seconds). Change all or few of the attributes on the DUT",
+                #          "Verify that the DUT sends reports for all the attributes that have changed after N seconds.")
+                ]
+
+    def on_notify_subscription_still_active(self):
+
+        self.previous_report_data_time = self.current_report_data_time
+        self.current_report_data_time = time.time()
+
+        if self.previous_report_data_time > 0:
+            diff = self.current_report_data_time - self.previous_report_data_time
+        else:
+            diff = 0
+
+        self.fprint(f"NotifyLogic {time.time()} >>> t: {diff}", "red")
+
+    def wait_for_attribute_update_report(self, expected_attribute, output):
+        try:
+            path, transaction = output.get(block=True, timeout=10)
+        except queue.Empty:
+            asserts.fail(
+                f"[AttributeChangeCallback | Local] Failed to receive a report for the {expected_attribute} attribute change")
+
+        asserts.assert_equal(path.AttributeType, expected_attribute,
+                             f"[AttributeChangeCallback | Local] Received incorrect report. Expected: {expected_attribute}, received: {path.AttributeType}")
+        try:
+            attribute_value = transaction.GetAttribute(path)
+
+            self.attr_update_report_data_time = time.time()
+
+            logging.info(
+                f"[AttributeChangeCallback | Local] Got attribute subscription report. Attribute {path.AttributeType}. Updated value: {attribute_value}. SubscriptionId: {transaction.subscriptionId}")
+        except KeyError:
+            asserts.fail("[AttributeChangeCallback | Local] Attribute {expected_attribute} not found in returned report")
+
+    current_report_data_time: time = 0
+    previous_report_data_time: time = 0
+    attr_update_report_data_time: time = 0
+
+    min_interval_floor_sec: int = 1
+    max_interval_ceiling_sec: int = 3
+
+    @async_test_body
+    async def test_TC_IDM_4_3(self):
+
+        # Test setup
+        node_label_attr = Clusters.BasicInformation.Attributes.NodeLabel
+        node_label_attr_path = [(0, node_label_attr)]
+        TH: ChipDeviceController = self.default_controller
+
+        # *** Step 1a ***
+        # DUT and TH activate the subscription.
+        self.step("1a")
+
+        # Subscribe to attribute
+        sub_th_step1ab = await TH.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(self.min_interval_floor_sec, self.max_interval_ceiling_sec),
+            keepSubscriptions=False
+        )
+
+        sub_th_step1ab.SetNotifySubscriptionStillActiveCallback(self.on_notify_subscription_still_active)
+
+        secs = 15
```

Comment:
why is this sleep here?

Reply:
Just testing out the intervals during n amount of seconds during development, not part of the test.